File: xchanger/src/main/java/com/finsim/xchanger/stocks/model/Stock.java
package com.finsim.xchanger.stocks.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "stocks")
public class Stock {
    @Id
    private String id;

    @Indexed(unique = true)
    private String isin;

    private String symbol;
    private String description;
    private String type;

    private StockPrices prices;

    public StockDto toDto() {
        return new StockDto(
            this.isin,
            this.symbol,
            this.description,
            this.type,
            this.prices
        );
    }
}

File: mainframe-lib/dossier/model/stock.go
package model

type Stock struct {
	Isin        string      `json:"isin"`
	Symbol      string      `json:"symbol"`
	Description string      `json:"description"`
	Type        string      `json:"type"`
	Prices      DailyPrices `json:"prices"`
}

type DailyPrices struct {
	DailyMax     Price `json:"dailyMax"`
	DailyMin     Price `json:"dailyMin"`
	DailyOpening Price `json:"dailyOpening"`
	DailyLast    Price `json:"dailyLast"`
}
